### PR TITLE
fix(remediation-service): add problemTitle to event message

### LINKED ISSUE #5548: Event message does not contain the `problemTitle`
##

* A remediation sequence gets triggered using: 
```
{
  "data": {
    "labels": {
      "Problem URL": "https:///#problems/problemdetails;pid=-240816790436828345_1633506659151V2"
    },
    "problem": {
      "ProblemTitle": "Http monitor global outage",
    },
    "project": "musicshop",
    "service": "carts",
    "stage": "production"
  },
  "id": "d485ea90-e1ef-4ce2-af21-32d44c3cb3fb",
  "source": "dynatrace-service",
  "specversion": "1.0",
  "time": "2021-10-06T07:53:26.637Z",
  "type": "sh.keptn.event.production.remediation.triggered",
  "shkeptncontext": "2d323430-3831-4637-b930-343336383238",
  "shkeptnspecversion": "0.2.1"
}
```

* The remediation.yaml looks as follows: 

```
apiVersion: spec.keptn.sh/0.1.4
kind: Remediation
metadata:
  name: carts-remediation
spec:
  remediations:
    - problemType: Response time degradation
      actionsOnOpen:
        - action: scaling
          name: scaling
          description: Scale up
          value: "1"
    - problemType: response_time_p90
      actionsOnOpen:
        - action: scaling
          name: scaling
          description: Scale up
          value: "1"
``` 

* The `problemTitle` does not match any `problemType` remediation.yaml. This is intended. 

## Bug

* However, I get the following message where I'm missing the `probelmTitle` for more information:

image

* An investigation into this situation depicted that just the `rootCause` is considered for writing this message: https://github.com/keptn/keptn/blob/master/remediation-service/handler/get_action_event_handler.go#L44

### Definition of Done
- [ ] An message is shown containing the `problemTitle` when available

### PR DESCRIPTION
**Pull request by bradmccoydev**
This PR closes #5548

- add problem title to event message 

Signed-off-by: Brad McCoy <bradmccoydev@gmail.com>

### CODE REVIEW

**Review by thisthat — DISMISSED** [review]
LGTM

**Comment by bradmccoydev** [comment]
@warber I have updated the GetNextAction function to cater to all scenarios and return an error that makes sense. Can you please review my suggested changes and give feedback.

File: remediation-service/handler/get_action_event_handler.go [hunk 1]
```diff
@@ -41,7 +42,7 @@ func (g *GetActionEventHandler) Execute(k sdk.IKeptn, event sdk.KeptnEvent) (int
 	// determine next action
 	action, err := GetNextAction(remediation, getActionTriggeredData.Problem, getActionTriggeredData.ActionIndex)
 	if err != nil {
-		return nil, &sdk.Error{Err: err, StatusType: keptnv2.StatusSucceeded, ResultType: keptnv2.ResultFailed, Message: "No more actions defined for problem type " + getActionTriggeredData.Problem.RootCause + " in remediation.yaml file."}
+		return nil, &sdk.Error{Err: err, StatusType: keptnv2.StatusSucceeded, ResultType: keptnv2.ResultFailed, Message: "No more actions defined " + err.Error() + " in remediation.yaml file."}
```
Comment (warber) [inline remediation-service/handler/get_action_event_handler.go:45]:
```suggestion
		return nil, &sdk.Error{Err: err, StatusType: keptnv2.StatusSucceeded, ResultType: keptnv2.ResultFailed, Message: err.Error()}
```

File: remediation-service/handler/get_action_event_handler.go [hunk 2]
```diff
@@ -115,12 +125,12 @@ func GetNextAction(remediation *v0_1_4.Remediation, problemDetails keptnv2.Probl
 
 	// we did not find an action
 	if actions == nil {
-		return nil, fmt.Errorf("unable to find actions for root cause %s", rootCause)
+		return nil, fmt.Errorf(problem)
```
Comment (warber) [inline remediation-service/handler/get_action_event_handler.go:128]:
```suggestion
		return nil, fmt.Errorf("unable to find action for %s", problem)
```

File: remediation-service/handler/get_action_event_handler.go [hunk 3]
```diff
@@ -115,12 +125,12 @@ func GetNextAction(remediation *v0_1_4.Remediation, problemDetails keptnv2.Probl
...
 	}
 
 	// the required action does not exist
 	if actionIndex >= len(actions) {
-		return nil, fmt.Errorf("failed to get action for root cause %s. There is no action with index %d", rootCause, actionIndex)
+		return nil, fmt.Errorf("%s. There is no action with index %d", problem, actionIndex)
```
Comment (warber) [inline remediation-service/handler/get_action_event_handler.go:133]:
```suggestion
		return nil, fmt.Errorf("there is no action with index %d for %s", actionIndex, problem)
```

File: remediation-service/handler/get_action_event_handler.go [hunk 4]
```diff
@@ -83,6 +84,15 @@ func ParseRemediationResource(resource *models.Resource) (*v0_1_4.Remediation, e
...
 	rootCause := problemDetails.RootCause
 	problemTitle := problemDetails.ProblemTitle
+	var problem string
+
+	if rootCause != "" {
+		problem = "root cause " + rootCause
+	} else if problemTitle != "" {
+		problem = "problem title " + problemTitle
+	} else {
+		problem = "root cause or problem title not found"
```
Comment (warber) [inline remediation-service/handler/get_action_event_handler.go:94]:
```suggestion
		problem = "problem type default"
```

**Comment by bradmccoydev** [comment]
Thanks @warber I have made the changes

**Review by warber — APPROVED** [review]

**Pull request merged**